### ISSUE TITLE
Multiple Motivations for using same Mathematical Composite Functional notation f(g(h(x))) for language syntax to encode Custom objective/loss/cost functions (XGBoost), as to encode Distributed Parallel Workflow Pipeline sequence.

### ISSUE THREAD

**Issue by Marc-Cox-08** [alan-turing-institute/MLJ.jl#488]
Several Motivations for using same Mathematical Composite Functional notation f(g(h(x)) for language syntax to encode Custom objective/loss/cost functions (XGBboost), as to encode Distributed Parallel Workflow Pipeline sequence.

( Mathematical Composite Functional notation / syntax to encode Custom-xgboost-loss-function , AND  also encode Distributed Parallel Workflow Pipeline sequence : )

Possibly in support of MLJ Roadmap here  https://github.com/alan-turing-institute/MLJ.jl/blob/master/ROADMAP.md , specifically 
**Broadening scope ( Generalize applicability )**

- Spin-off a stand-alone measures (loss functions) package (currently here)
- AD-powered gradient descent methods

Feature Request Issue: Need PR specification describing design considerations motivated by Julia Discourse discussion link here >> https://discourse.julialang.org/t/custom-xgboost-loss-function-w-zygote-julia-computing-blog-post/35811/23

Thank You for the Linux/Unix Pipe code which is a very helpful foundation and a natural fit for technically underpinning distributed parallel processing using Distributed. I believe we can continue to elevate the Julia language syntax at a higher abstraction layer using mathematical notation, while simultaneously keeping the high performance Automatic vectorization gears hidden/encapsulated to achieve this >> https://en.wikipedia.org/wiki/Automatic_vectorization

So this PR request includes Julia language support syntax for Function Composition (computer science) described here >> 
https://en.wikipedia.org/wiki/Function_composition_%28computer_science%29

"The ability to easily compose functions encourages factoring (breaking apart) functions for maintainability and code reuse. More generally, larger systems might be built by composing whole programs."

Generally allowing recomposable resequencable function calls via RE-Composible Functional Programming is desirable.  For example I am presently rewriting alot of my basically Julia procedural / .jl script code into functions and function calls so that I can easily resequence the order of operations so that when AutoML determines that f(g(h(x)) is less effective for machine learning cross validation etc. than h(g(f(x))) reordering the sequence of operations will be intuitive and so easily accomplished by the developer (supervised, semi-supervised ML), or the program itself for AutoML (automatic unsupervised ML) .

Generally we see computer language syntax/mathematics notation as critical to facilitate expressivity via mathematical expressions – which is a major benefit of writing Julia code for scientific computing, and actually at the heart of this thread discussion subject , here >>  "Custom XGBoost Loss function w/ Zygote. Julia Computing blog post" << because Cross Validation of Zygote Loss functions will involve mathematics notation such as in terms of the derivatives of f'(x) and g'(x) and the product of functions as follows:  f(g(x))' =  f'(g(x)) * g'(x)
per https://en.wikipedia.org/wiki/Chain_rule

Quick examples:

Current proposed Unix Pipe syntax:
	using Distributed
	nprocs() == 1 && addprocs()
	@everywhere using AutoMLPipeline, DataFrames
	#Get models.
sk= AutoMLPipeline.SKLearners.learner_dict |> keys |> collect;

Request we upfit/override operator Pipe syntax To Proposed Function composition / Functional Programming pseudo-code syntax:
	using Distributed
	nprocs() == 1 && addprocs()
#Get models.
sk= collect(keys(AutoMLPipeline.SKLearners.learner_dict))

Also we note that it may be that the Functional Programming pseudo-code syntax above already (almost?) works but even so I believe it is still important for us Julia developers to get in the habit of writing Functional Composition example code to support Calculus multiple derivative notations e.g. f(g(x))' =  f'(g(x)) * g'(x) so we can most easily write the XGBoost custom loss functions here as other boosting methods do, where they **generalize them by allowing optimization of an _arbitrary differentiable loss function_** as per >> https://en.wikipedia.org/wiki/Gradient_boosting

Best Regards,
-Marc Cox

Additional Motivation for PR Request> I believe we'll have to get the math notation compact and expressive as possible to hide/encapsulate complexity inside functions before approaching "How to define a Fitness Function in a Genetic Algorithm ?"  

One description of Genetic Algos here NOTE: defined by strictly mathematical syntax >> 
https://www.mathworks.com/help/gads/examples/coding-and-minimizing-a-fitness-function-using-the-genetic-algorithm.html

and "How to define a Fitness Function in a Genetic Algorithm ?" here >>
 https://towardsdatascience.com/how-to-define-a-fitness-function-in-a-genetic-algorithm-be572b9ea3b4

Additional basis in MLJ code base for PR Request> 
In MLJ a pipeline is a composite model here >> https://alan-turing-institute.github.io/MLJ.jl/dev/composing_models/

-AND Here>> https://alan-turing-institute.github.io/MLJ.jl/dev/mlj_cheatsheet/#Pipelines-1
